app/src/main/java/org/p2p/wallet/send/ui/main/SendFragment.kt

```diff
@@ -145,79 +142,53 @@ class SendFragment :
                 toolbar.setNavigationIcon(R.drawable.ic_back)
                 toolbar.setNavigationOnClickListener { popBackStack() }
             }
-            sendButton.setOnClickListener { presenter.sendOrConfirm() }
 
-            targetTextView.setOnClickListener {
-                addFragment(SearchFragment.create())
-            }
-            targetImageView.setOnClickListener {
-                addFragment(SearchFragment.create())
-            }
-            messageTextView.setOnClickListener {
-                addFragment(SearchFragment.create())
-            }
+            targetTextView.setOnClickListener { addFragment(SearchFragment.create()) }
+            targetImageView.setOnClickListener { addFragment(SearchFragment.create()) }
+            messageTextView.setOnClickListener { addFragment(SearchFragment.create()) }
 
-            clearImageView.setOnClickListener {
-                presenter.setTargetResult(null)
-            }
+            clearImageView.setOnClickListener { presenter.setTargetResult(result = null) }
 
-            AmountFractionTextWatcher.installOn(amountEditText) {
-                presenter.setNewSourceAmount(it)
-            }
+            AmountFractionTextWatcher.installOn(amountEditText) { presenter.setNewSourceAmount(it) }
+            val originalTextSize = amountEditText.textSize
+            // Use invisible auto size textView to handle editText text size
+            amountEditText.doOnTextChanged { text, _, _, _ -> handleAmountTextChanged(text, originalTextSize) }
```

Comment:
you can user .afterTextChanged {}, which has one required parameter

Reply:
sry, i missed your comment and merged it already :(